vsgenstubs4/init.py

```diff
@@ -0,0 +1,579 @@
+
+
+implementation_start = '# implementation'
+implementation_end = '# end implementation'
+
+
+class Implementation(NamedTuple):
+    plugin: PluginMeta
+    content: List[str]
+
+    @classmethod
+    def from_namespace(cls, namespace: str, cores: Sequence[CoreLike]) -> 'Implementation':
+        return Implementation(PluginMeta.from_namespace(namespace, cores), [])
+
+    @staticmethod
+    def get_name(plugin: PluginMeta, core_name: str, /) -> str:
+        return f'_Plugin_{plugin.name}_{core_name}_Bound'
+
+    def __gt__(self, x: 'Implementation', /) -> bool: return self.plugin.__gt__(x.plugin)  # type: ignore[override]
+    def __lt__(self, x: 'Implementation', /) -> bool: return self.plugin.__lt__(x.plugin)  # type: ignore[override]
+    def __ge__(self, x: 'Implementation', /) -> bool: return self.plugin.__ge__(x.plugin)  # type: ignore[override]
+    def __le__(self, x: 'Implementation', /) -> bool: return self.plugin.__le__(x.plugin)  # type: ignore[override]
+
+
+def make_implementations(plugins: Iterable[PluginMeta]) -> Iterator[Implementation]:
+    for plugin in plugins:
+        implementation_content = chain.from_iterable(
+            (
+                '',
+                f"class {Implementation.get_name(plugin, core_name)}(Plugin):",
+                '    """'
+                f'This class implements the module definitions for the "{plugin.name}" VapourSynth plugin.'
+                '\\n\\n*This class cannot be imported.*'
+                '"""',
+                '\n'.join(signatures),
+            ) for core_name, signatures in plugin.bound
+        )
+
+        content = chain.from_iterable([
+            ['', f"{implementation_start}: {plugin.name}"],
+            implementation_content,
+            ['', implementation_end, '']
+        ])
+
+        yield Implementation(plugin, list(content))
+
+
+instance_start = '# instance_bound_'
+instance_end = '# end instance'
+
+instance_bound_pattern = re.compile(fr"^{instance_start}([^:]+): (.+)")
+
+
+class Instance(NamedTuple):
+    plugin: PluginMeta
+    core_name: str
+    definition: List[str]
+
+    @classmethod
+    def from_namespace(cls, namespace: str, core_name: str, cores: Sequence[CoreLike]) -> 'Instance':
+        return Instance(PluginMeta.from_namespace(namespace, cores), core_name, [])
+
+    @staticmethod
+    def get_head(plugin: PluginMeta, core_name: str) -> str:
+        return f"{instance_start}{core_name}: {plugin.name}"
+
+    def __gt__(self, x: 'Implementation', /) -> bool: return self.plugin.__gt__(x.plugin)  # type: ignore[override]
+    def __lt__(self, x: 'Implementation', /) -> bool: return self.plugin.__lt__(x.plugin)  # type: ignore[override]
+    def __ge__(self, x: 'Implementation', /) -> bool: return self.plugin.__ge__(x.plugin)  # type: ignore[override]
+    def __le__(self, x: 'Implementation', /) -> bool: return self.plugin.__le__(x.plugin)  # type: ignore[override]
```

Comment:
[Why not use functools.total_odering?](https://docs.python.org/3/library/functools.html#functools.total_ordering)

Reply:
https://github.com/vapoursynth/vsrepo/pull/186#discussion_r973697159